cmd/mapt/cmd/azure/hosts/rhel.go

```diff
@@ -0,0 +1,141 @@
+
+			// ParseEvictionRate
+			var spotToleranceValue = spotAzure.DefaultEvictionRate
+			if viper.IsSet(paramSpotTolerance) {
+				var ok bool
+				spotToleranceValue, ok = spotAzure.ParseEvictionRate(
+					viper.GetString(paramSpotTolerance))
+				if !ok {
+					return fmt.Errorf("%s is not a valid spot tolerance value", viper.GetString(azparams.ParamSpotTolerance))
+				}
+			}
+			instanceRequest := &instancetypes.AzureInstanceRequest{
+				CPUs:       viper.GetInt32(params.CPUs),
+				MemoryGib:  viper.GetInt32(params.Memory),
+				Arch:       util.If(viper.GetString(params.LinuxArch) == "arm64", instancetypes.Arm64, instancetypes.Amd64),
+				NestedVirt: viper.GetBool(params.NestedVirt),
+			}
+
+			// Initialize gh actions runner if needed
+			if viper.IsSet(params.InstallGHActionsRunner) {
+				err := ghactions.InitGHRunnerArgs(viper.GetString(params.GHActionsRunnerToken),
+					viper.GetString(params.GHActionsRunnerName),
+					viper.GetString(params.GHActionsRunnerRepo))
+				if err != nil {
+					logging.Error(err)
+				}
+			}
+
+			if err := azureRHEL.Create(
+				&azureRHEL.Request{
+					Prefix:               viper.GetString(params.ProjectName),
+					Location:             viper.GetString(paramLocation),
+					VMSizes:              viper.GetStringSlice(paramVMSize),
+					InstanceRequest:      instanceRequest,
+					Version:              viper.GetString(paramLinuxVersion),
+					Arch:                 viper.GetString(params.LinuxArch),
+					SubsUsername:         viper.GetString(params.SubsUsername),
+					SubsUserpass:         viper.GetString(params.SubsUserpass),
+					ProfileSNC:           viper.IsSet(params.ProfileSNC),
+					Username:             viper.GetString(paramUsername),
+					Spot:                 viper.IsSet(paramSpot),
+					SpotTolerance:        spotToleranceValue,
+					SetupGHActionsRunner: viper.IsSet(params.InstallGHActionsRunner)}); err != nil {
+				logging.Error(err)
+			}
+			return nil
+		},
+	}
+	flagSet := pflag.NewFlagSet(params.CreateCmdName, pflag.ExitOnError)
+	flagSet.StringP(params.ConnectionDetailsOutput, "", "", params.ConnectionDetailsOutputDesc)
+	flagSet.StringToStringP(params.Tags, "", nil, params.TagsDesc)
+	flagSet.StringP(paramLocation, "", defaultLocation, paramLocationDesc)
+	flagSet.StringP(params.LinuxArch, "", params.LinuxArchDefault, params.LinuxArchDesc)
+	flagSet.StringSliceP(paramVMSize, "", []string{}, paramVMSizeDesc)
+	flagSet.StringP(paramLinuxVersion, "", defaultRHELVersion, paramLinuxVersionDesc)
+	flagSet.StringP(paramUsername, "", defaultUsername, paramUsernameDesc)
+	flagSet.StringP(params.SubsUsername, "", "", params.SubsUsernameDesc)
+	flagSet.StringP(params.SubsUserpass, "", "", params.SubsUserpassDesc)
+	flagSet.Bool(paramSpot, false, paramSpotDesc)
+	flagSet.StringP(paramSpotTolerance, "", defaultSpotTolerance, paramSpotToleranceDesc)
+	flagSet.AddFlagSet(params.GetGHActionsFlagset())
+	flagSet.AddFlagSet(params.GetCpusAndMemoryFlagset())
```

Comment:
the `flagSet` is missing the `params.ProfileSNC` flag

Reply:
or this will be done in a follow up PR? and this is handling only the refactoring to make it easier to re-use the profile code for azure

Reply:
No no totally missed it, actually I did try this out but it ended up failing need to debug it a bit more.

But thx for the catch may it was related to this